reVX/plexos/rev_reeds_plexos.py

```diff
@@ -465,9 +571,25 @@ def _parse_plexos_nodes(plexos_nodes):
         plexos_nodes : pd.DataFrame
             Plexos node meta data including gid, lat/lon, voltage
         """
-        plexos_nodes = parse_table(plexos_nodes)
-        plexos_nodes = DataCleaner.pre_filter_plexos_meta(plexos_nodes)
+
+        if (isinstance(plexos_nodes, str)
+                and plexos_nodes.endswith(('.csv', '.json'))):
+            plexos_nodes = parse_table(plexos_nodes)
+        elif isinstance(plexos_nodes, str) and plexos_nodes.endswith('.shp'):
+            rc = RegionClassifier(self.sc_build, plexos_nodes,
+                                  regions_label=None)
+            plexos_nodes = rc._regions
+            if 'plexos_id' not in plexos_nodes:
+                plexos_nodes['plexos_id'] = np.arange(len(plexos_nodes))
+
+        elif not isinstance(plexos_nodes, pd.DataFrame):
+            msg = ('Did not recognize plexos_nodes input: {}'

```

Comment:
We should make the error message more specific, i.e. expecting x, y, or z, but got:

Reply:
Yep can do!